### PR TITLE
Add lerna and yarn workspaces

### PR DESCRIPTION
**Pull request by mrice32**
Right now, these are just alternative ways of running `npm install`:
```
yarn
```
or 
```
lerna bootstrap
```

This will allow us to slowly add packages that will be managed by lerna.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
   "name": "protocol",
   "version": "0.1.0",
   "description": "UMA Protocol Smart Contracts",
```
Comment (chrismaree) [inline package.json:4]: I know not related to this PR but is this description still accurate? should be `UMA protocol Mono repo` perhaps?

Comment (mrice32) [reply package.json:4]: Yeah, I'll send out a related PR to do some general updates on package.json files.

File: package.json [hunk 2]
```diff
@@ -2,6 +2,7 @@
...
+  "private": true,
```
Comment (nicholaspai) [inline package.json:5]: Why did you change this to private?

Comment (mrice32) [reply package.json:5]: Yarn workspaces require it for the root `package.json`: https://classic.yarnpkg.com/en/docs/workspaces/#toc-how-to-use-it.

File: .gitignore [hunk 3]
```diff
@@ -2,6 +2,7 @@ node_modules
 build
 bridge.log
 package-lock.json
+yarn.lock
```
Comment (adrianmcli) [inline .gitignore:5]: The lock file should be checked in: https://classic.yarnpkg.com/en/docs/yarn-lock#toc-check-into-source-control

**Comment by adrianmcli** [comment]
nit: you don't need to type `yarn install`, you can just type `yarn`.

File: .gitignore (hunk 3 above)
Comment (mrice32) [reply .gitignore:5]: `npm` has the same sort of lock file (`package-lock.json`), but we've always left it out. It causes issues with merging/rebasing, and it pollutes diffs with tons of unimportant changes. See [here](https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5#comment78284962_44210813) for similar complaints. Many [major](https://github.com/jquery/jquery) [projects](https://github.com/serverless/serverless) leave it out.

We've never had any serious problems leaving it out. The only issues were those caused by us choosing not to lock down versions in the `package.json` (which `package-lock.json` kinda forces you to do anyway). If your package only works with certain versions, that can be specified in the `package.json` by locking down the package. We should be specifying those versions explicitly if we worry that certain versions break our code.

There are some other technical complaints about how you're locking in versions in a pretty arbitrary way that package maintainers didn't intend: https://dev.to/gajus/stop-using-package-lock-json-or-yarn-lock-3ddi.

Happy to discuss this more, but I think we should leave this issue for a separate thread/PR since our repo currently operates without `package-lock.json`.

**Comment by mrice32** [comment]
> nit: you don't need to type `yarn install`, you can just type `yarn`.

oooo, thanks. That's very convenient.

**Comment by mrice32** [comment]
> > nit: you don't need to type `yarn install`, you can just type `yarn`.
> 
> oooo, thanks. That's very convenient.

Updated in the description.
